var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -89,50 +90,100 @@ class Julia(MakefilePackage):
+        # 'USE_SYSTEM_FFTW=1',
+        # 'USE_SYSTEM_GMP=1',
+        # 'USE_SYSTEM_MPFR=1',
+        # 'USE_SYSTEM_PCRE=1',
         options = [
-            "USE_BINARYBUILDER=0",
-            "USE_SYSTEM_LLVM={0}".format(1 if spec.satisfies('+external_llvm') else 0),
-            "USE_SYSTEM_PCRE=1",
-            "USE_SYSTEM_LIBM=1",
-            # "USE_SYSTEM_BLAS=1",
-            # "USE_SYSTEM_LAPACK=1",
-            "USE_SYSTEM_GMP=1",
-            "USE_SYSTEM_MPFR=1",
-            "USE_SYSTEM_SUITESPARSE=1",
-            # "USE_SYSTEM_LIBUV=1",
-            "USE_SYSTEM_CURL=1",
-            "USE_SYSTEM_LIBGIT2=1",
-
-            # START Can be removed in future versions!
-            "override USE_BINARYBUILDER_LLVM=0",
-            "override USE_BINARYBUILDER_PCRE=0",
-            "override USE_BINARYBUILDER_OPENBLAS=0",
-            "override USE_BINARYBUILDER_OPENLIBM=0",
-            "override USE_BINARYBUILDER_SUITESPARSE=0",
-            "override USE_BINARYBUILDER_MBEDTLS=0",
-            "override USE_BINARYBUILDER_LIBSSH2=0",
-            "override USE_BINARYBUILDER_CURL=0",
-            "override USE_BINARYBUILDER_LIBGIT2=0",
-            "override USE_BINARYBUILDER_PCRE=0",
-            "override USE_BINARYBUILDER_LIBUV=0",
-            "override USE_BINARYBUILDER_UNWIND=0",
-            # END remove
-
-            # "# LIBBLAS={0}".format(spec["blas"].libs),
-            # "LIBBLASNAME={0}".format(spec["blas"].name),
-            # "# LIBLAPACK={0}".format(spec["lapack"].libs),
-            # "LIBLAPACKNAME={0}".format(spec["lapack"].name),
-            "prefix={0}".format(prefix)
+            'prefix={0}'.format(prefix)
         ]
+        if '@:0.5.0' in spec:
+            options += [
+                'override USE_SYSTEM_CURL=1'
+            ]
+
+        if '+cxx' in spec:
+            options += [
+                'BUILD_LLVM_CLANG=1',
+                'LLVM_ASSERTIONS=1',
+                'USE_LLVM_SHLIB=1'
+            ]
+        if spec.target.family == 'aarch64':
+            options += [
+                'JULIA_CPU_TARGET=generic',
+                'MARCH=armv8-a+crc'
+            ]
+
+        if spec.target.family == 'x86_64' or spec.target.family == 'x86':
+            if spec.target == 'x86_64':
+                options += [
+                    'JULIA_CPU_TARGET=generic'
+                ]
+            else:
+                options += [
+                    'JULIA_CPU_TARGET={0}'.format(spec.target)
+                ]
+
+        if '%intel' in spec:
+            options += [
+                'USEICC=1',
+                'USEIFC=1',
+                'USE_INTEL_LIBM=1'
+            ]
+
+        if '+mkl' in spec:
+            options += [
+                'USE_INTEL_MKL=1',
+            ]
         with open('Make.user', 'w') as f:
             f.write('\n'.join(options) + '\n')
+
+        make()
+        make('install')
+
+        # Julia's package manager needs a certificate
```

Comment:
Interesting!

Reply:
> I assume this is copied from upstream? Then I won’t bother about nitpicks for commented options/code.
> 
> Did you check the module? May be we can ask Polina to test pulls module?

Yes, yes, and maybe we don't need to. I pointed her to the module here in the ticket. The script she provided runs fine with the module.